src/Advection/weno_fifth_order.jl

```diff
@@ -361,9 +361,64 @@ end
 @inline right_biased_α₂(FT, ψ, T, scheme, args...) = scheme.C3₀ / (right_biased_β₂(FT, ψ, T, scheme, args...) + FT(ε))^ƞ
 
 #####
-##### VectorInvariant reconstruction (based on JS or Z)
+##### VectorInvariant reconstruction (based on JS or Z) (z-direction Val{3} is different from x- and y-directions)
 #####
 
+@inline function left_biased_weno5_weights(FT, ijk, T, scheme::WENOVectorInvariant, ::Val{3}, idx, loc, u)
+    i, j, k = ijk
+
+    u₂, u₁, u₀ = left_stencil_z(i, j, k, u)
+    
+    β₀ = left_biased_β₀(FT, u₀, T, scheme, Val(3), idx, loc)
+    β₁ = left_biased_β₁(FT, u₁, T, scheme, Val(3), idx, loc)
+    β₂ = left_biased_β₂(FT, u₂, T, scheme, Val(3), idx, loc)
+
+    if scheme isa ZWENOVectorInvariant
+        τ₅ = abs(β₂ - β₀)
+        α₀ = scheme.C3₀ * (1 + (τ₅ / (β₀ + FT(ε)))^ƞ) 
+        α₁ = scheme.C3₁ * (1 + (τ₅ / (β₁ + FT(ε)))^ƞ) 
+        α₂ = scheme.C3₂ * (1 + (τ₅ / (β₂ + FT(ε)))^ƞ) 
+    else
+        α₀ = scheme.C3₀ / (β₀ + FT(ε))^ƞ
+        α₁ = scheme.C3₁ / (β₁ + FT(ε))^ƞ
+        α₂ = scheme.C3₂ / (β₂ + FT(ε))^ƞ
+    end
```

Comment:
Can we reduce copy/paste in this file?

While we're in here we could also resolve some of #2271